great_expectations/datasource/fluent/spark_file_path_datasource.py

```diff
@@ -36,9 +36,36 @@ def _get_reader_options_include(self) -> set[str] | None:
         return {"header", "infer_schema"}
 
 
+class ParquetAsset(_FilePathDataAsset):
+    type: Literal["parquet"] = "parquet"
+    # The options below are available for parquet as of spark v3.4.0
+    # See https://spark.apache.org/docs/latest/sql-data-sources-parquet.html for more info.
+    datetime_rebase_mode: Literal["EXCEPTION", "CORRECTED", "LEGACY"] = Field(
+        alias="datetimeRebaseMode"
+    )
+    int_96_rebase_mode: Literal["EXCEPTION", "CORRECTED", "LEGACY"] = Field(
+        alias="int96RebaseMode"
+    )
+    merge_schema: bool = Field(False, alias="mergeSchema")
+
+    class Config:
+        extra = pydantic.Extra.forbid
+        allow_population_by_field_name = True
+
+    def _get_reader_method(self) -> str:
+        return self.type
+
+    def _get_reader_options_include(self) -> set[str] | None:
+        """These options are available as of spark v3.4.0
+
+        See https://spark.apache.org/docs/latest/sql-data-sources-parquet.html for more info.
+        """
+        return {"datetimeRebaseMode", "int96RebaseMode", "mergeSchema"}
+
+
 class _SparkFilePathDatasource(_SparkDatasource):
     # class attributes
-    asset_types: ClassVar[List[Type[DataAsset]]] = [CSVAsset]
+    asset_types: ClassVar[List[Type[DataAsset]]] = [CSVAsset, ParquetAsset]
 
     # instance attributes
     assets: List[_FilePathDataAsset] = []  # type: ignore[assignment]
```

Comment:
```suggestion
    assets: List[Union[CSVAsset, ParquetAsset]] = []  # type: ignore[assignment]
```

This will give us a better schemas and runtime validation.
The `assets` can't be any `_FilePathDataAsset` type, they will only ever be a `CSVAsset`, or `ParquetAsset`

Reply:
Good call! I ended up putting it in a variable so I could reference elsewhere: `_SPARK_FILE_PATH_ASSET_TYPES` - happy to rename if there is a better name.